de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java

```diff
@@ -110,20 +110,23 @@ private SaveMetadataCallback(SaveAsDialog save_dialog) {
         }
 
         @Override
-         public void onFailure(Throwable caught) {
-             save_dialog.hide();
-             announcer.schedule(new ErrorAnnouncementConfig("Unable to save your file. Please try again or contact support."));
-         }
+        public void onFailure(Throwable caught) {
+            save_dialog.unmask();
+            if (caught.getMessage().contains("ERR_EXISTS")) {
+                announcer.schedule(new ErrorAnnouncementConfig(appearance.fileExistsError()));
```

Comment:
Is it possible to get the filename that the user entered from the `SaveAsDialog` here?

If so, then you could use the existing `org.iplantc.de.resources.client.messages.IplantErrorStrings#fileExists` error message, which takes a list of 1 or more file names to include with the message, instead of adding this new `fileExistsError` message.

Although I did notice that the `fileExists` message is inserting a `<br />` tag between the message and the filename(s), so I'm not sure how that would look with the `ErrorAnnouncementConfig`.